modin/backends/pandas/query_compiler.py

```diff
@@ -507,6 +507,26 @@ def transpose(self, *args, **kwargs):
         # Switch the index and columns and transpose the data within the blocks.
         return self.__constructor__(self._modin_frame.transpose())
 
+    def columnarize(self):
+        """
+        Transposes this QueryCompiler if it has a single row but multiple columns.
+
+        This method should be called for QueryCompilers representing a Series object,
```

Comment:
Currently, `columnarize` method is called in Series init function only, where we already know data has a proper shape. Before creating Series object we use `is_series_like` to check the shape. Both functions are used to move index accesses to a backend and provide an alternate implementation in the OmniSci backend, which would be able to avoid index access in certain cases.

Reply:
I see, thanks!